pyproject.toml

```diff
@@ -17,11 +17,18 @@ readme = "README.md"
 dynamic = ["version"]
 
 dependencies = [
```

Comment:
I would avoid being so strict with this pinnings. we are making people's life unnecessarily complicated if they want to use the library. Sure, it may require a bit more work from our side in terms of bugs, but I don't think that using semver 3.0.2 vs >=3.0 will make that much of a difference...

Maybe:
Pydantic > 2.7 (since we actually require this one)
SemVer > 3.x
I will leave the others up to you since i don't know the dependencies

Reply:
Updated with >=